### ISSUE TITLE
draft pre-flight check plugin baked into draft init

### ISSUE THREAD

**Issue by michelleN** [Azure/draft#306]
TBD: name for the plguin

Draft is built on other modular components to help developers create apps that will end up on Kubernetes. In some cases, you'll already have some dependencies. It'd be nice for `draft init` to not only check for these dependencies but also start or run installation steps if necessary. So for example, if minikube is not running, then start it. If Tiller is not installed, run the `helm init` command. We as a team have talked about creating a plugin that checks for the dependencies and configures them if they are not already running/installed. Then, we'd like to make that a "default" plugin that runs as part of the `draft init` command after the client side steps in the `draft init` logic.

**Comment by bacongobbler** [comment]
I wonder if this pre-flight plugin should also handle the case where we want to install certain plugins when `draft init` is called too, for example the ongoing work in #2. Would that make sense for this plugin?

**Comment by bacongobbler** [comment]
I have been thinking a bit about

> I wonder if this pre-flight plugin should also handle the case where we want to install certain plugins when draft init is called too, for example the ongoing work in #2

And have been looking into how other projects like docker are doing this. This sort of workflow isn't actually enforced at the tooling level, but at the packaging layer. If you fetch the latest .pkg of docker, it will install all of the "blessed" plugins into your $PATH. This gives vendors the power to distribute different packages with a completely different set of plugins, all set and ready to go for their platform.

I'm wondering if we should do the same here for this plugin: we can bake in the default `draft pre-flight` and `draft pack-repo` plugins into our distributed packages at https://github.com/Azure/draft/releases/latest, and then tie them into `helm init` but just warn if those plugins aren't present, giving vendors the option to provide their own pre-flight and pack-repo plugins (or not provide them at all if it's irrelevant). "Batteries included, but removable" rings a bell here.

What do you think about that?

**Comment by michelleN** [comment]
Yeah I like that experience. If I'm not mistaken, this is what we talked about a few days ago in our meeting. Given that in the default experience that the user doesn't have to even know those are plugins, I'm 100% on board. 👍

**Comment by bacongobbler** [comment]
Yep! Just commenting this publicly for posterity (also because I tend to forget things over time). :)

**Comment by bacongobbler** [comment]
I still think a pre-flight check plugin would be useful, though the architecture has significantly changed since we opened this ticket.

I'll leave it up to Michelle on whether or not we want to close this ticket based on the current architecture. :)
